libraries/ESP8266HTTPClient/examples/DigestAuthorization/DigestAuthorization.ino

```diff
@@ -0,0 +1,133 @@
+/*
+    This sketch shows how to handle HTTP Digest Authorization.
+
+    Written by Parham Alvani and Sajjad Rahnama, 2018-01-07.
+
+   This example is released into public domain,
+   or, at your option, CC0 licensed.
+ */
+
+#include <ESP8266WiFi.h>
+
+#include <ESP8266HTTPClient.h>
+
+const char* ssid = "........";
+const char* ssidPassword = "........";
+
+const char *username = "admin";
+const char *password = "admin";
+
+const char *server = "http://httpbin.org";
+const char *uri = "/digest-auth/auth/admin/admin/MD5";
+
+String exractParam(String& authReq, const String& param, const char delimit){
+  int _begin = authReq.indexOf(param);
+  if (_begin==-1) return "";
+  return authReq.substring(_begin+param.length(),authReq.indexOf(delimit,_begin+param.length()));
+}
+
+String getCNonce(const int len) {
+    static const char alphanum[] =
+        "0123456789"
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz";
+    String s = "";
+
+    for (int i = 0; i < len; ++i) {
+        s += alphanum[rand() % (sizeof(alphanum) - 1)];
+    }
+
+    return s;
+}
+
+String getDigestAuth(String& authReq, const String& username, const String& password, const String& uri, int counter) {
+  // extracting required parameters for RFC 2069 simpler Digest
+  String realm = exractParam(authReq, "realm=\"", '"');
+  String nonce = exractParam(authReq, "nonce=\"", '"');
+  String cNonce = getCNonce(8);
+
+  char nc[8];
+  sprintf(nc, "%08d", counter);
```

Comment:
If that's "%08d", the size of `nc` should be 9 bytes to accommodate the zero terminator.

Reply:
Counter is an int, so nc should have size 12. Then, the format specifier should be %11d.
Counter should actually be an unsigned int imo, it doesn't make sense for negative values. In that case, nc should have size 11.

Reply:
Ok, let's read the [RFC](https://tools.ietf.org/html/rfc2617):

     The nc-value is the hexadecimal
     count of the number of requests (including the current request)
     that the client has sent with the nonce value in this request.  For
     example, in the first request sent in response to a given nonce
     value, the client sends "nc=00000001".

and

     nc-value         = 8LHEX

So it must be "%08x".

Reply:
yes, you are right, it must be in hex format so I convert it into `%08x`.

Reply:
Agreed. But what stresses that counter should be unsigned :P

Reply:
@devyte I corrects that too :)